src/StrictPhp/StrictPhpKernel.php

```diff
@@ -43,13 +43,152 @@
 use StrictPhp\TypeChecker\TypeChecker\TypedTraversableChecker;
 use StrictPhp\TypeFinder\PropertyTypeFinder;
 
-class StrictPhpKernel extends AspectKernel
+/**
+ * Note: not extensible on purpose. This is a bootstrapper/kernel that is totally procedural.
+ *
+ * Overriding
+ */
+final class StrictPhpKernel extends AspectKernel
 {
+    const CHECK_STATE_AFTER_CONSTRUCTOR_CALL   = 'CHECK_STATE_AFTER_PUBLIC_METHOD_CALL';
+    const JAIL_PUBLIC_METHOD_PARAMETERS        = 'JAIL_PUBLIC_METHOD_PARAMETERS';
+    const CHECK_STATE_AFTER_PUBLIC_METHOD_CALL = 'CHECK_STATE_AFTER_PUBLIC_METHOD_CALL';
+    const CHECK_PUBLIC_METHOD_PARAMETER_TYPE   = 'CHECK_PUBLIC_METHOD_PARAMETER_TYPE';
+    const CHECK_PUBLIC_METHOD_RETURN_TYPE      = 'CHECK_PUBLIC_METHOD_RETURN_TYPE';
+    const CHECK_PROPERTY_WRITE_IMMUTABILITY    = 'CHECK_PROPERTY_WRITE_IMMUTABILITY';
+    const CHECK_PROPERTY_WRITE_TYPE            = 'CHECK_PROPERTY_WRITE_TYPE';
+
     /**
      * {@inheritDoc}
      */
     protected function configureAop(AspectContainer $container)
     {
+        // empty: configuration should come from elsewhere
+    }
+
+    /**
+     * @param mixed[]  $options  {@see \Go\Core\AspectKernel::init()}
+     * @param string[] $features
+     *
+     * @return self
+     */
+    public static function bootstrap(
+        $options = [],
+        $features = [
+            self::CHECK_STATE_AFTER_CONSTRUCTOR_CALL,
```

Comment:
Not sure what features to enable/disable by default, but right now I'd say that following should be left out by default:

```php
self::JAIL_PUBLIC_METHOD_PARAMETERS,
self::CHECK_PROPERTY_WRITE_IMMUTABILITY,
self::CHECK_PROPERTY_WRITE_TYPE,
```

These are probably too strict and have serious limitations, currently caused by:

 - insability of recognizing `instanceof` calls
 - limitations in go-aop-php (public properties support)